From b6acba75fc183d8494865f24e79461d8e46cc403 Mon Sep 17 00:00:00 2001
From: qwang2 <quanyang.wang@windriver.com>
Date: Thu, 5 Jul 2018 21:45:40 +0800
Subject: [PATCH 3/8] nxp-imx6: change pcie power-domain property from pd_pci
 to pd_display

There are 4 power-domains (CPU, GPU, DISPLAY, PCIE-PHY) in i.mx6sx. PCIe
controller in i.mx6sx depends on display relative clocks
(IMX6SX_CLK_LVDS1_OUT, IMX6SX_CLK_DISPLAY_AXI etc) and on pcie phy which
is included in PCIE-PHY power domain. When we enter "mem" power state,
pci bus will hang when accessing register of pci config space because DISPLAY
power domain is off previously. So we need attach pci controller device
to DISPLAY power domain to avoid this.

We also need enable PCIE-PHY power domain because during booting, if no device
is attched to it, it will be off and pci's phy will not work.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6sx.dtsi |    2 +-
 drivers/soc/imx/gpc.c         |    1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index 93f906d..471ed5f 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -1626,7 +1626,7 @@
 				 <&clks IMX6SX_CLK_DISPLAY_AXI>;
 			clock-names = "pcie", "pcie_bus", "pcie_phy", "pcie_inbound_axi";
 			pcie-phy-supply = <&reg_pcie_phy>;
-			power-domains = <&pd_pci>;
+			power-domains = <&pd_disp>;
 			fsl,max-link-speed = <2>;
 			status = "disabled";
 		};
diff --git a/drivers/soc/imx/gpc.c b/drivers/soc/imx/gpc.c
index 57b0169..06ea67b 100644
--- a/drivers/soc/imx/gpc.c
+++ b/drivers/soc/imx/gpc.c
@@ -410,6 +410,7 @@ static int imx_pgc_power_domain_remove(struct platform_device *pdev)
 		},
 		.reg_offs = 0x200,
 		.cntr_pdn_bit = 6,
+		.flags = PGC_DOMAIN_FLAG_NO_PD,
 	},
 };
 
-- 
1.7.9.5

